rsocket/framing/FramedReader.cpp

```diff
@@ -131,9 +131,7 @@ void FramedReader::parseFrames() {
 
 void FramedReader::onComplete() {
   payloadQueue_.move();
-  if (DuplexConnection::Subscriber::subscription()) {
```

Comment:
I believe we fixed a bug with this check.
Subscriber depends on having a Subscription otherwise it crashes.

Would you mind running a server and trying to connect this server with a non-rsocket client and observe if the server is crashing?

Reply:
I agree with Alex, that if somebody is calling onComplete, they have called onSubscribe and we should have subscription. Maybe we should turn it into DCHECK(DuplexConnection::Subscriber::subscription()) ?
Let's find the problem Fuat is describing, maybe the fix he is talking about should have been in another place, not here.

Reply:
@phoad It's in the summary

> Connecting to RSocket server with nc and typing garbage still works properly.

@lehecka That exact DCHECK is already done by `DuplexConnection::Subscriber::onComplete()`